packages/core/src/fhir-deduplication/resources/diagnostic-report.ts

```diff
@@ -92,23 +97,65 @@ export function groupSameDiagnosticReports(diagReports: DiagnosticReport[]): {
     const datetime = getDateFromResource(diagReport, "datetime");
     const isPresentedFormPresent = diagReport.presentedForm?.length;
     const isResultPresent = diagReport.result?.length;
-    if (datetime && (isPresentedFormPresent || isResultPresent)) {
-      const key = JSON.stringify({ datetime });
-      fillMaps(
-        diagReportsMap,
-        key,
-        diagReport,
+
+    // If a diagnostic report does not contain any results or doctor's notes, it is useless and can be removed
+    if (!isPresentedFormPresent && !isResultPresent) {
+      danglingReferences.add(createRef(diagReport));
+      continue;
+    }
+
+    const practitionerRefsSet = new Set<string>();
+
+    const getterKeys: string[] = [];
+    const setterKeys: string[] = [];
+
+    diagReport.performer?.forEach(perf => {
+      const ref = perf.reference;
+      if (ref) {
+        if (ref.includes("Practitioner")) {
+          practitionerRefsSet.add(ref);
+          return;
+        }
+      }
+    });
+
+    const practitionerRefs = Array.from(practitionerRefsSet).map(p => ({ practitioner: p }));
+    if (datetime && practitionerRefs.length > 0) {
+      setterKeys.push(...createKeysFromObjectArray({ datetime }, practitionerRefs));
+      setterKeys.push(...createKeysFromObjectAndFlagBits({ datetime }, [1]));
+
+      getterKeys.push(...createKeysFromObjectArray({ datetime }, practitionerRefs));
+      getterKeys.push(...createKeysFromObjectAndFlagBits({ datetime }, [0]));
+    }
+
+    if (datetime && practitionerRefs.length === 0) {
+      setterKeys.push(...createKeysFromObjectAndFlagBits({ datetime }, [0]));
+
+      getterKeys.push(...createKeysFromObjectAndFlagBits({ datetime }, [1]));
```

Comment:
This is the opposite from what we do if length > 0, expected?

Also, worth adding a comment maybe to explain what we're doing here?

Reply:
was expected. I removed the bits from this dedup cuz I no longer think it is necessary with the current logic